server/action_creators.js

```diff
@@ -70,6 +71,24 @@ export function addOrder(entries) {
   };
 }
 
+export function changeEntryStatus(orderId, entryIndex, status) {
+  return (dispatch, getState) => {
+    const orders = getState().orders;
+    const orderIndex = orders.findIndex((order) => order.id === orderId);
+
+    return Order.updateEntry(orderId, entryIndex, { status })
+      .then((response) => {
+        const updatedOrders = [
+          ...orders.slice(0, orderIndex),
+          response[0],
+          ...orders.slice(orderIndex + 1)
+        ];
+
+        dispatch(setState({ orders: updatedOrders }));
```

Comment:
Yeah looking into this.. Reason UI doesn't currently reflect, I think, is that I've only set the `defaultValue` of `<select>`. Need to set `value` to be `this.props.status` but not sure why app state updates aren't getting pushed to the `Entry` component props...